x-pack/plugins/secops/public/components/page/hosts/authentications_table/index.tsx

```diff
@@ -234,8 +234,14 @@ const getAuthenticationColumns = (startDate: number): Array<Columns<Authenticati
                 <DragEffects>
                   <Provider dataProvider={dataProvider} />
                 </DragEffects>
+              ) : isNil(get('lastFailure.host.id', node)) ? (
+                <>{hostName}</>
               ) : (
-                hostName
+                <EuiLink
+                  href={`#/link-to/hosts/${encodeURIComponent(node.lastFailure!.host!.id!)}`}
+                >
+                  {hostName}
+                </EuiLink>
```

Comment:
You could almost make a specific host based higher order component that uses the `<EuiLink` to reduce boiler plate if you see in our patterns. Kind of the beauty of React is that it supports reductionism well.

For example something like below (spit balling here),

```ts
<HostLink host={get(lastFailure.host)} />
```

And then within that HOC, you know you have a hosts ECS object that is typed to a host object. If you need to check if that host is completely empty and return `<>--</>` through `getEmptyTag()` you can. If the host object exists but it doesn't have an id for some reason, then you would just return a plain hostName without a link to the host page.

Reply:
Thanks for the feedback! This is indeed ripe for some abstraction at this point. I could see a few components around routing to facilitate behaviors like this and keep things clean. I'll open a ticket to track this lower hanging tech debt.